src/utils/xhr-utils.js

```diff
@@ -136,11 +137,18 @@ export function fromStructuredCloneable(response, responseType) {
 
   const isDocumentType = responseType == 'document';
   if (!isDocumentType) {
+    let body = response['body'];
+    // Handle JSON response.
+    if (body && body['json']) {
```

Comment:
Just curious: who's nesting a `json` field under `body` today? AFAICT Gmail isn't using this. We just serialize the body (which can only be `application/json` today) into a string and put it directly into the `body` field.

Reply:
This is just following the way json is accessed via API .https://developer.mozilla.org/en-US/docs/Web/API/Body/json. We can go your route as well as only application/json is supported but I wanted to do this incase of future support for other data formats.

Reply:
Removed nested json field as header content-type can just be used to read format types in the future. Thanks Su.